packages/starlight/__tests__/remark-rehype/asides.test.ts

```diff
@@ -143,3 +143,22 @@ test('runs without locales config', async () => {
 	const res = await processor.render(':::note\nTest\n::');
 	expect(res.code.includes('aria-label=Note"'));
 });
+
+test('tranforms back unhandled text directives', async () => {
+	const res = await processor.render(
+		`This is a:test of a sentence with a text:name[content]{key=val} directive.`
+	);
+	expect(res.code).toMatchInlineSnapshot(`
+		"<p>This is a:test
+		 of a sentence with a text:name[content]{key="val"}
+		 directive.</p>"
+	`);
```

Comment:
Noting a subtle difference here in the transformation back to text where the attribute value becomes wrapped in `"`. It should be extremely rare that this is an issue though, as it only applies in a relatively complex string that is a correctly formed directive, so I guess that’s fine?

```suggestion
		`This is a:test of a sentence with a text:name[content]{key=val} directive.`
	);
	expect(res.code).toMatchInlineSnapshot(`
		"<p>This is a:test
		 of a sentence with a text:name[content]{key="val"}
		 directive.</p>"
	`);
```

Reply:
Excellent catch, I totally missed that.

After a bit of digging, I found that `mdast-util-directive` respects [`options.quote`](https://github.com/syntax-tree/mdast-util-to-markdown#optionsquote) altho in our case, this would not be enough to fix the issue.

I think the current approach is a good enough solution for now but if we ever hit the rare case you mentioned where we cannot restore the node to its exact original content including the quotes, we would need to revisit this. I think the 2 options would be to either:

- Only restore the node if it's a directive with no attributes and document/emit a warning when encountering a case where we cannot restore (this seems to be the Docusaurus approach)
- Find another way to properly restore the node (which I think would mean to have a copy of the AST before the directive was transformed)

Are you still okay with this approach?

- If no, I can work a refactor based on the solution we would choose
- If yes, I can merge this PR as it contains no doc changes in its current state

Reply:
Yes, 100% approve — we can merge and release this as is.

If we run into this rare issue, we could also look into an alternative to `remark-directive` that _only_ provides the container directive and not also text/lead directives.